spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/DashboardHelperService.java

```diff
@@ -176,6 +177,41 @@ public JSONObject searchProject(HttpSession session, JSONObject json) {
     }
   }
 
+
+  public JSONObject searchPool(HttpSession session, JSONObject json) {
+    String searchStr = json.getString("str");
+    try {
+      List<Pool> pools;
+      StringBuilder b = new StringBuilder();
+      if (!isStringEmptyOrNull(searchStr)) {
+
+        pools = new ArrayList<Pool>(requestManager.listAllPoolsBySearch(searchStr));
+      } else {
+        pools = new ArrayList<Pool>(requestManager.listAllPoolsWithLimit(50));
+      }
+
+      if (pools.size() > 0) {
+        Collections.sort(pools);
+        Collections.reverse(pools);
+        for (Pool p : pools) {
+          // yuck-o
+          b.append("<a class=\"dashboardresult\" href=\"/miso/pool/" + p.getId()
+              + "\"><div onMouseOver=\"this.className=&#39dashboardhighlight&#39\" onMouseOut=\"this.className=&#39dashboard&#39\" class=\"dashboard\">");
```

Comment:
I'm very confused about `&#39`. Should there be a `;` at the end? Also, doesn't `JSONObjectResponse` escape the string? If so, doesn't that mean `'` is legal?